pkg/metrics/metrics.go

```diff
@@ -84,10 +87,23 @@ var ACMEClientRequestDurationSeconds = prometheus.NewSummaryVec(
 	[]string{"scheme", "host", "path", "method", "status"},
 )
 
+type registeredCertificates struct {
+	certificates map[*v1alpha1.Certificate]struct{}
```

Comment:
I don't think we can use the pointer value of the certificate here to identify it, as it *could* change (the listers don't make guarantees here afaik).

There are already tools in client-go for handling this sort of thing, and we utilise them for our own controller workqueues as well. You can see some docs here: https://godoc.org/k8s.io/client-go/tools/cache#DeletionHandlingMetaNamespaceKeyFunc

This will ultimately convert the API types to a string `namespace/name`, which can then be keyed upon 😄